### PR TITLE
Add `conversion/doc-to-text` profile

### PR DESCRIPTION
**Pull request by janhalama**
## Description

This PR adds `conversion/doc-to-text` profile along with `cloudmersive` provider

## Motivation and Context

This profile is used in CV Analyzer example app.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: grid/conversion/doc-to-text/profile.supr [hunk 1]
```diff
@@ -0,0 +1,108 @@
+"""
+Convert document to text
+Convert's document to plain text.
```
Comment (jnv) [inline grid/conversion/doc-to-text/profile.supr:3]: typo: 
```suggestion
Convert a document to plain text.
```

File: grid/conversion/doc-to-text/profile.supr [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+"""
+name = "conversion/doc-to-text"
+version = "1.0.0"
+
+"""
+Convert Document to Text
+Convert document's in various formats (pdf, doc, rtf) to plain text.
```
Comment (jnv) [inline grid/conversion/doc-to-text/profile.supr:10]: typo:

```suggestion
Convert documents in various formats (pdf, doc, rtf) to plain text.
```

File: grid/conversion/doc-to-text/profile.supr [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    Content
+    Binary content of the document
+    """
+    content!
+  }
+
+  result {
+    """
+    Text
+    The document converted to plain text.
```
Comment (jnv) [inline grid/conversion/doc-to-text/profile.supr:29]: suggestion:
```suggestion
    The document contents as plain text.
```

File: grid/conversion/doc-to-text/profile.supr [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  """
+  title! string!
+  """
+  Detail
+  A human-readable explanation specific to this occurrence of the problem.
+  """
+  detail string! 
+  """
+  Code
+  Error code.
```
Comment (jnv) [inline grid/conversion/doc-to-text/profile.supr:72]: idea: I'm missing some explanation that it's intended for the programmer, or do you think it's redundant to state that?

**Review by jnv — CHANGES_REQUESTED** [review]
Beside a few typos all other comments are non-blocking.

I have one suggestion though: would you consider changing the scope to `file-conversion`, for example? Conversion is an overloaded term, my first idea was "what this has to do with user conversion?"

File: grid/conversion/doc-to-text/maps/cloudmersive.test.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  ) => {
+    recordings.forEach(
+      (recording: {
+        path: string;
+        method?: string | undefined;
+        body?: RequestBodyMatcher | undefined;
+      }) => {
+        if (
+          recording.method === 'POST' &&
+          recording.path === '/convert/autodetect/to/txt'
```
[the remark this answers is not in the document]
Comment (janhalama) [reply grid/conversion/doc-to-text/maps/cloudmersive.test.ts:22]: Yes that would be better, but request headers are not recorded by default (as they contain dynamic values preventing the matcher to match requests), and AFAIK nock allows either record all of them or none.

File: .env.example [hunk 6]
```diff
@@ -5,6 +5,7 @@ AZURE_MAPS_SUBSCRIPTION_KEY=xxx
 AZURE_SPEECH_TO_TEXT_SUBSCRIPTION_KEY=xxx
 BITBUCKET_PASSWORD=xxx
 BITBUCKET_USERNAME=xxx
+CLOUDMERSIVE_API_KEY=xxx
```
[the remark this answers is not in the document]
Comment (janhalama) [reply .env.example:8]: That is the convention in Station for now. Security variables and integration parameters are not grouped together. I would stick with the convention here and fix as a separate PR.

[the remark this answers is not in the document]
Comment (jnv) [reply .env.example:8]: Ok, just that the second parameter was at the end of the file. Sorry, I highlighted the wrong line.

File: grid/conversion/doc-to-text/maps/doc-to-text.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    describe('ConvertDocumentToText', () => {
+      it('should return error when converting empty file', async () => {
+        const result = await superface.run(
+          {
+            useCase: 'ConvertDocumentToText',
+            input: {
+              fileName: 'empty-file.pdf',
+              content: BinaryData.fromPath(
+                'grid/conversion/doc-to-text/maps/test-files/empty-file.pdf'
```
[the remark this answers is not in the document]
Comment (janhalama) [reply grid/conversion/doc-to-text/maps/doc-to-text.ts:30]: sure, thanks for the suggestion

[the remark this answers is not in the document]
Comment (janhalama) [reply grid/conversion/doc-to-text/maps/doc-to-text.ts:30]: fixed

File: grid/conversion/doc-to-text/maps/cloudmersive.test.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        ) {
+          const boundaryReplacement = '---boundary replaced';
+          //Boundary contains random value, we are replacing it with constant to match the recording
```
[the remark this answers is not in the document]
Comment (janhalama) [reply grid/conversion/doc-to-text/maps/cloudmersive.test.ts]: 👍  I created standalone multipartFormDataBoundaryMatcherHook helper hook.

**Comment by janhalama** [comment]
> Beside a few typos all other comments are non-blocking.
> 
> I have one suggestion though: would you consider changing the scope to `file-conversion`, for example? Conversion is an overloaded term, my first idea was "what this has to do with user conversion?"

That makes sense. I changed the scope `conversion` to proposed `file-conversion`.

File: grid/conversion/doc-to-text/profile.supr (hunk 4 above)
Comment (janhalama) [reply grid/conversion/doc-to-text/profile.supr]: I added the explanation.

File: .env.example (hunk 6 above)
[the remark this answers is not in the document]
Comment (janhalama) [reply .env.example:8]: Yes I see now. I fixed the ordering of integration parameters.

**Review by jnv — APPROVED** [review]
LGTM!

**Pull request merged**